virtcontainers/qemu.go

```diff
@@ -584,6 +586,13 @@ func (q *qemu) createSandbox(ctx context.Context, id string, networkNS NetworkNa
 		return err
 	}
 
+	// Add PCIe Root Port devices to hypervisor
+	// The pcie.0 bus do not support hot-plug, but PCIe device can be hot-plugged into PCIe Root Port.
+	// For more details, please see https://github.com/qemu/qemu/blob/master/docs/pcie.txt
+	if hypervisorConfig.HypervisorMachineType == QemuQ35 && hypervisorConfig.PCIeRootPort > 0 {
```

Comment:
`hypervisorConfig.HypervisorMachineType == QemuQ35` is not needed since you're validating it in `genericAppendPCIeRootPort`